src/main/java/com/mraof/minestuck/item/RightClickMessageItem.java

```diff
@@ -0,0 +1,44 @@
+package com.mraof.minestuck.item;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.*;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+
+public class RightClickMessageItem extends Item
+{
+	
+	public RightClickMessageItem(Properties properties)
+	{
+		super(properties);
+	}
+	
+	@Override
+	public ActionResult<ItemStack> onItemRightClick(World worldIn, PlayerEntity playerIn, Hand handIn)
+	{
+		if(worldIn.isRemote)
+		{
+			if(playerIn.getHeldItem(handIn).getItem() == MSItems.EIGHTBALL)
```

Comment:
I prefer that you don't write functionality that checks for specific items like this. Try using multiple classes, or some sort of parameter to the constructor

Reply:
Done!

Reply:
For this purpose I wouldn't say that an int is an appropriate parameter type. An enum will beter fit this usage.